tests/playwright/src/specs/kind.spec.ts

```diff
@@ -0,0 +1,86 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import { type Page } from '@playwright/test';
+import { expect as playExpect } from '@playwright/test';
+import { afterAll, beforeAll, beforeEach, describe, test } from 'vitest';
+
+import { ExtensionsPage } from '../model/pages/extensions-page';
+import { ResourcesPage } from '../model/pages/resources-page';
+import { WelcomePage } from '../model/pages/welcome-page';
+import { NavigationBar } from '../model/workbench/navigation';
+import { StatusBar } from '../model/workbench/status-bar';
+import { PodmanDesktopRunner } from '../runner/podman-desktop-runner';
+import type { RunnerTestContext } from '../testContext/runner-test-context';
+import { isCI, isLinux } from '../utility/platform';
+import { waitForPodmanMachineStartup } from '../utility/wait';
+
+let pdRunner: PodmanDesktopRunner;
+let page: Page;
+let navigationBar: NavigationBar;
+let resourcePage: ResourcesPage;
+let statusBar: StatusBar;
+const extensionLabel: string = 'podman-desktop.kind';
+
+beforeAll(async () => {
+  pdRunner = new PodmanDesktopRunner();
+  page = await pdRunner.start();
+  pdRunner.setVideoAndTraceName('kind-e2e');
+  const welcomePage = new WelcomePage(page);
+  await welcomePage.handleWelcomePage(true);
+  await waitForPodmanMachineStartup(page);
+  navigationBar = new NavigationBar(page);
+  resourcePage = new ResourcesPage(page);
+  statusBar = new StatusBar(page);
+});
+
+beforeEach<RunnerTestContext>(async ctx => {
+  ctx.pdRunner = pdRunner;
+});
+
+afterAll(async () => {
+  await pdRunner.close();
+});
+
+describe('Kind End-to-End Tests', async () => {
+  test.skipIf(isCI && isLinux)('Install Kind CLI if Not Present', async () => {
+    await navigationBar.openSettings();
+    if ((await resourcePage.ResourceExists('kind').isVisible({ timeout: 5000 })) === false && !isLinux) {
+      await statusBar.installKindCLI();
+    }
+  });
+  test('Verify Kind CLI Installation', async () => {
+    await navigationBar.openSettings();
+    await playExpect(statusBar.getKindInstallationButton()).not.toBeVisible({ timeout: 5000 });
+    await playExpect(resourcePage.ResourceExists('kind')).toBeVisible({ timeout: 5000 });
+  });
+  test('Explore and manage Kind extension lifecycle', async () => {
+    const extensionsPage = new ExtensionsPage(page);
+    await navigationBar.openExtensions();
+    await playExpect.poll(async () => await extensionsPage.extensionIsInstalled(extensionLabel)).toBeTruthy();
+    await playExpect(extensionsPage.getExtension(extensionLabel).getByText('ACTIVE')).toBeVisible();
```

Comment:
There is a method in the POM that returns the status of an extension, you do not need to manually grab the status field in the test, in general handling the web elements themselves inside the actual tests should be avoided as much as possible since it defeats the purpose of using POM